src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5801,9 +5802,19 @@ private static BoundExpression CreatePlaceholderIfNecessary(BoundExpression expr
                 TakeIncrementalSnapshot(node); // Visit does this before visiting each node
                 TypeWithState receiverType = visitAndCheckReceiver(node);
 
+                receiver = null;
                 while (true)
                 {
-                    ReinferMethodAndVisitArguments(node, receiverType);
+                    VisitArgumentResult? extensionReceiverResult = null;
+                    if (receiver is not null && node.ReceiverOpt is null)
```

Comment:
>if (receiver is not null && node.ReceiverOpt is null) [](http://example.com/codeflow?start=20&length=53)

I think it is much harder to follow the logic when we are handling instance receivers in one place (below, when we are preparing for the next iteration), and handling extension receivers in another place (here, at the beginning of the iteration). Because of that we also need to check ```receiver is not null``` and do ```receiver = null;``` before the loop, so that we can skip the first iteration. Consider handling receivers in one place, below where instance receivers are handled, by adding ```else``` part for ```if (node.ReceiverOpt is not null)```.